crates/network-primitives/src/traits.rs

```diff
@@ -67,6 +67,18 @@ pub trait HeaderResponse {
 
     /// Blob fee for the next block (if EIP-4844 is supported)
     fn next_block_blob_fee(&self) -> Option<u128>;
+
+    /// Miner/Coinbase of the block
+    fn miner(&self) -> Address;
+
+    /// Gas limit of the block
+    fn gas(&self) -> u128;
+
+    /// Mix hash of the block
+    fn mix_hash(&self) -> Option<B256>;
+
+    /// Difficulty of the block
+    fn difficulty(&self) -> U256;
```

Comment:
maybe we don't need this, I assume this could eventually be removed

Reply:
OP stack based L2s use `block.difficulty` for generating random values.

Reply:
But I guess `block.prevrandao` serves the same purpose on L2s